### DISCUSSION TITLE
sempy_labs with administrative API

### DISCUSSION THREAD

**Discussion by dtiles** [microsoft/semantic-link-labs#188] [Ideas]
Hi,
we're experimenting a lot with Fabric and the API's. I've found a lot of potential in the sempy_labs, especially for Power BI Administrators.
One issue that I see is that sempy_labs requires the user to have access to the workspace directly. A great addition would be to be for a Power BI Tenant Administrator to have access to the API's results without needing to be enabled to the workspace.
For example, in this way, it would be possible to search all the semantic models that connect to an old deprecated schema, for example.

**Comment by m-kovalsky** [comment]
All the functions in the [admin subpackage](https://semantic-link-labs.readthedocs.io/en/stable/sempy_labs.admin.html) are Power BI / Fabric admin APIs which are meant for the tenant admin.

https://learn.microsoft.com/rest/api/power-bi/admin

https://learn.microsoft.com/rest/api/fabric/admin/domains

**Reply by dtiles** [reply]
Hi,
yes, we're already using the Admin API. With that, we can get all the datasets of a tenant.
Unfortunately, it seems that sempy_labs works only with direct access to the workspace, which means that a workflow like this is not possible (unless you say otherwise):

1. Get all the workspaces and semantic models of a Power BI Tenant
2. For each semantic model, connect with sempy_labs to get some information. This connection doesn't require the user (a Power BI Tenant Administrator) to have direct access to the workspace and semantic model

**Reply by m-kovalsky** [reply]
Your comment is in line with similar requests from other folks (#171). We're looking to expand the capabilities for so called 'tenant-scans'.

**Comment by m-kovalsky** [comment]
See [0.9.0](https://github.com/microsoft/semantic-link-labs/releases/tag/0.9.0).
